### ISSUE TITLE
SetDragRectangles doesn't work with changing rects

### ISSUE THREAD

**Issue by jamers99** [microsoft/WindowsAppSDK#2953]
I've created this issue in this repo to bring more visibility. It's a duplicate of microsoft/microsoft-ui-xaml#7392 I'm not actually sure which repo I should be reporting it in. As noted below, this is **extremally** important to our application that we're migrating (100's of customers and growing to 1000's).

### Describe the bug

`AppWindow.TitleBar.SetDragRectangles` works well for the most part, clunky as it is, it's pretty straightforward.

However, when you need to change the rectangles (UI changed in title bar) there is a bug that it doesn't completely update.

### Steps to reproduce the bug

1. Create an app that has interactive content that may change in size in the title bar.
2. Make on the interactive content `SizeChanged` event, calculate the Rect and call `SetDragRectangles`
3. Run the app
4. Change the interactive content size
5. Try to interact in the new area that would have previously been part of the grip
**It doesn't interact with the element.** _It doesn't even grip... it just blocks interaction. If you right click it pops the window menu._

Here is a simple project reproducing the issue: [TitleBarTest.zip](https://github.com/microsoft/microsoft-ui-xaml/files/9134501/TitleBarTest.zip)

### Expected behavior

It should fully update and I should be able to interact with my control everywhere, including in the new space it took up.

### Screenshots

Here is a video describing the issue.
https://user-images.githubusercontent.com/20806279/179582055-4fe6d676-a3e2-4e24-9ddf-6b5a9ebb0d24.mp4

### NuGet package version

WinUI 3 - Windows App SDK 1.1.2

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

Windows 11 (21H2): Build 22000

### Additional context

This is incredibly important to our app. We have browser style tabs in our title bar.

**Comment by riverar** [comment]
Can confirm this isn't working correctly. `AppWindow.TitleBar.ResetToDefault();` before calling `SetDragRectangles` does seem to workaround the issue but creates very annoying flicker.

Heads up @rkarman @BenJKuhn

**Comment by jamers99** [comment]
I was aware of that "workaround" and am currently trying to implement that in a non-jarring way. I'm doubtful I'll be able to get it completely clean, but hoping to at least get the content of the window not to move around. Will update if I find something. REALLY wishing for open source right now.....

**Comment by riverar** [comment]
Seems related to #2574 and appears to be fixed in Windows App SDK 1.2 Preview 1. (I upgraded your sample project and it's working now, can you verify?)

**Comment by jamers99** [comment]
Nice! I verified the fix.

But, since we need this app to work on Windows 10, I've actually headed down the route of using `Window.SetTitleBar`. The issue with that API is that **it doesn't support multiple zones**. 

Well, I thought, it's not like we need support two pointers gripping at once... _why don't we just switch the grip zone on hover?_
**That worked!** So given that there doesn't _seem_ to be a downside, I think we're going to go with that solution until this hot mess of windowing API's gets improved (I _really_ don't like how complex the Win11 AppWindow API is). 

Hopefully will be resolved with [this road map item](https://portal.productboard.com/winappsdk/1-windows-app-sdk/c/64-improved-default-and-custom-title-bar-via-windowing-apis)...
